src/theory/arith/nonlinear_extension.cpp

```diff
@@ -2508,6 +2492,61 @@ void NonlinearExtension::check(Theory::Effort e) {
 
     } while (needsRecheck);
   }
+
+  // Did we internally determine a model exists? If so, we need to record some
+  // information in the theory engine's model class.
+  if (d_builtModel.get())
+  {
+    if (e == Theory::EFFORT_FULL)
+    {
+      // don't need to build the model yet
+      return;
+    }
```

Comment:
what about the other effort calls?

Reply:
Effort here is either FULL or LAST_CALL, where we only need to build a model if we are at LAST_CALL. I'll make the condition here more inclusive (if the effort is less than LAST_CALL.

Reply:
I see